mlflow/models/evaluation/base.py

```diff
@@ -1782,6 +1782,12 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
             if isinstance(model, _ServedPyFuncModel):
                 os.kill(model.pid, signal.SIGTERM)
 
+        # if model_id is specified log metrics to the eval run and logged model
+        if model_id is not None:
+            from mlflow import log_metrics
+
+            log_metrics(metrics=evaluate_result.metrics, dataset=data, model_id=model_id)
```

Comment:
nit -- i think we use this pattern more frequently
```suggestion
            mlflow.log_metrics(metrics=evaluate_result.metrics, dataset=data, model_id=model_id)
```